lib/dictionaryanalyzer.rb

```diff
@@ -0,0 +1,120 @@
+require 'pry'
+require 'yaml'
+
+class DictionaryAnalyzer
+  attr_accessor :return_values
+  def initialize(dictionary_input)
+    @current_dictionary = dictionary_input
+    word_count(dictionary_input)
+    letter_data(dictionary_input)
+    @return_values = []
+    @match_counter = 0
+  end
+
+  def word_count(input)
+    answer = input.length
+    puts "There are #{answer} words in this dictionary.\n"
+  end
+
+  def letter_data(input)
+    ("a".."z").each do |regex|
+      regex_big = regex.upcase
+      letter_count = (input.join("\n").scan(/^#{regex}/).size + input.join("\n").scan(/^#{regex_big}/).size) #regex hates me letter_counter_helper(input, regex) #
+      puts "There are #{letter_count} words under the letter #{regex}"
+    end
+  end

```

Comment:
There is an easier way to do this, using string index! Reference the hash key with `letter_count[string[0]]`.